FABulous.py

```diff
@@ -0,0 +1,744 @@
+        specObject = self.fabricGenerator.generateBitsStreamSpec()
+        return specObject
+
+
+class FABulousShell(cmd.Cmd):
+    intro: str = f"""
+
+    ______      ____        _
+    |  ____/\   |  _ \      | |
+    | |__ /  \  | |_) |_   _| | ___  _   _ ___
+    |  __/ /\ \ |  _ <| | | | |/ _ \| | | / __|
+    | | / ____ \| |_) | |_| | | (_) | |_| \__ \\
+    |_|/_/    \_\____/ \__,_|_|\___/ \__,_|___/
+
+
+Welcome to FABulous shell
+You have started FABlous shell with following options:
+{' '.join(sys.argv[1:])}
+
+Type help or ? to list commands
+The shell support tab completion for commands and files
+
+To run the complete FABulous flow, type 
+    load_fabric <fabric_file>
+    run_FABulous
+    """
+    prompt: str = "FABulous> "
+    fabricGen: FABulous
+    projectDir: str
+    top: str
+    allTile: List[str]
+    csvFile: str
+    extension: str = "v"
+    fabricLoaded: bool = False
+
+    def __init__(self, fab: FABulous, projectDir: str):
+        super().__init__()
+        self.fabricGen = fab
+        self.projectDir = projectDir
+        self.tiles = []
+        self.superTiles = []
+        self.csvFile = ""
+        if isinstance(self.fabricGen.writer, VHDLWriter):
+            self.extension = "vhdl"
+        else:
+            self.extension = "v"
+
+        if hasattr(fab, "fabric"):
+            self.fabricLoaded = True
+
+    def precmd(self, line: str) -> str:
+        if "gen" in line or "run" in line and not self.fabricLoaded:
+            logger.error("Fabric not loaded")
+            return ""
+        return line
+
+    # override the emptyline method, so empty command will just do nothing
+    def emptyline(self):
+        pass
+
+    def parse(self, line: str) -> List[str]:
+        return line.split()
+
+    def _complete_path(self, path):
+        if os.path.isdir(path):
+            return glob(os.path.join(path, "*"))
+        else:
+            return glob(path + "*")
+
+    def _complete_tileName(self, text):
+        return [t for t in self.allTile if t.startswith(text)]
+
+    def do_exit(self, args):
+        "exit FABlous shell"
+        logger.info("Exiting FABulous shell")
+        return True
+
+    def do_load_fabric(self, args):
+        "load csv file and generate an internal representation of the fabric"
+        logger.info("Loading fabric")
+        args = self.parse(args)
+        # if no argument is given will use the one set by set_fabric_csv
+        # else use the argument
+        if len(args) == 0:
+            if self.csvFile != "" and os.path.exists(self.csvFile):
+                self.fabricGen.loadFabric(self.csvFile)
+            else:
+                logger.error(
+                    "No argument is given and no csv file is set or the file does not exist")
+                return
+        else:
+            self.fabricGen.loadFabric(args[0])
+        self.fabricLoaded = True
+        self.csvFile = args[0]
+        self.allTile = list(self.fabricGen.fabric.tileDic.keys())
```

Comment:
this is now failing in a new way for supertiles: for DSPs, for example, my understanding is that this should just contain `DSP`. But actually it now ends up containing `DSP_top` for example, failing with the following backtrace:

```
  File "/home/gatecat/FABulous-refactor/FABulous.py", line 745, in <module>
    fabShell.cmdloop()
  File "/usr/lib/python3.10/cmd.py", line 138, in cmdloop
    stop = self.onecmd(line)
  File "/usr/lib/python3.10/cmd.py", line 217, in onecmd
    return func(arg)
  File "/home/gatecat/FABulous-refactor/FABulous.py", line 372, in do_run_FABulous
    self.do_gen_fabric()
  File "/home/gatecat/FABulous-refactor/FABulous.py", line 343, in do_gen_fabric
    self.do_gen_all_tile()
  File "/home/gatecat/FABulous-refactor/FABulous.py", line 337, in do_gen_all_tile
    self.do_gen_tile(self.allTile)
  File "/home/gatecat/FABulous-refactor/FABulous.py", line 273, in do_gen_tile
    if subTiles := [f.name for f in os.scandir(f"./Tile/{t}") if f.is_dir()]:
FileNotFoundError: [Errno 2] No such file or directory: './Tile/DSP_top'
```

because the correct path it should be scanning is `./Tile/DSP`